### ISSUE TITLE
New change to Stripe.js has caused this to break

### ISSUE THREAD

**Issue by rglynn-dev** [ectoflow/vue-stripe-js#31]
I believe Stripe has made a new change to their Stripe.js hosted at https://js.stripe.com/v3, which causes this library to error. In production it just errors to the console and doesn't break, but in development/test it prevents rendering elements entirely.

The error thrown by Stripe.js is `Elements cannot be mounted in a ShadowRoot. Please mount in the Light DOM.`.

I believe this is an accurate error, because it is not mounting the Stripe elements correctly.
```
/src/components/StripeElement.vue:53

  stripeElement.value = createElement(
     elements.value,
     type.value,
     options.value
    )
   stripeElement.value.mount(domElement.value)   <-------- domElement not yet attached to the dom
   mountPoint.value.appendChild(domElement.value)
 ```
 
 If I change the order to append to the DOM first:
```diff
 const mountElement = () => {
+       mountPoint.value.appendChild(domElement.value)
        stripeElement.value = createElement(
          elements.value,
          type.value,
          options.value
        )
        stripeElement.value.mount(domElement.value)
-       mountPoint.value.appendChild(domElement.value)
      }
```

The error is gone and everything works as expected.

I'm not sure if this is the most appropriate fix for the issue, but would be happy to raise a PR.

Edit:
Better diff formatting

**Comment by ibox4real** [comment]
We are seeing similar issues as described.

**Comment by mohamedriyas7373** [comment]
![image](https://github.com/user-attachments/assets/84ed76c3-9dc7-4b42-a7e5-0c782b572c2e)

We are getting the same issue in production. Please fix the issue immediately.

**Comment by mohamedali850** [comment]
Hey! What was the update that caused this issue? We are unable to collect payments. Please respond to this ASAP!

**Comment by softbeehive** [comment]
Folks, I’ll take a look when I land in around two hours. There were no changes in vue-stripe-js recently. Two possible reasons Stripe change or Vue update

It’s unfortunate situation but I’d like to remind those who DEMAND an immediate fix, this is open source project, so be respectful

**Comment by mohamedali850** [comment]
Thanks @softbeehive We appreciate your words!

**Comment by mbeneyto** [comment]
Hi, we are having the same problem with the deprecated library for Vue 2. In our case we are still using Nuxt 2, so I doubt it's a problem derived from a Vue change.

Also, if we make the same change proposed by @rglynn-dev it seems to work fine.
https://github.com/ectoflow/vue-stripe-elements/blob/master/src/components/StripeElement.vue#L76

```diff
mountElement() {
       const domEl = this.domElement
       this.stripeElement = createElement(this.elements, this.type, this.options)
-     this.stripeElement.mount(domEl)
       this.$el.appendChild(domEl)
+     this.stripeElement.mount(domEl)
 
       return this.stripeElement
}
```

**Comment by mohamedriyas7373** [comment]
Thanks @rglynn-dev, I tried your solution, and it is working fine.

```
node_modules\vue-stripe-js\dist\vue-stripe.es.js

const mountElement = () => {
        stripeElement.value.mount(domElement.value);
        mountPoint.value.appendChild(domElement.value);
        stripeElement.value = createElement(elements.value, type.value, options.value);
      };
```

**Comment by michaeloconn98** [comment]
I've found that I don't get the issue with the card input not rendering in production environments but I do in non production environments. For users on production the error just gets reported in their consoles but the functionality still works just fine for them from what I've seen.

Needless to say the fix suggested by @rglynn-dev does appear to fix the issue. Once I did the fix on my local the input renders in and I am able to use it as normal from there.

Currently on version 1.0.1 of the package.

**Comment by code2prog** [comment]
Same story ;)

**Comment by mbeneyto** [comment]
I've contacted Stripe via their developers Discord and they have identified the issue and should be solved today. In any case, it might be worth having a look at @rglynn-dev fix and append the dom element before mounting the Stripe Element.

cc @softbeehive

**Comment by softbeehive** [comment]
Alright, as a hotfix I'll change the order as suggested by @rglynn-dev, thanks! Tested locally, will investigate what changed on Stripe side.

**Comment by softbeehive** [comment]
**vue-stripe-js@1.0.3** has been published on npm. Tested in with vite

```json
"vue-stripe-js": "1.0.3",
```

**Comment by michaeloconn98** [comment]
@softbeehive much appreciated console error has gone for me and input is now rendering in on my local environment. I am about to test it for next couple hours and release to my production env some point soon. 

Will update you if anything breaks that I notice.

**Comment by softbeehive** [comment]
https://github.com/stripe/stripe-js/issues/662 is closed. I haven’t seen any reasonable explanation from Stripe. Might be related to how stripe.js works with iframes (specualtion). If you have suffered a downtime, I encourage you as a customer to request the detailed postmortem and share your findings here.

Closing the issue

**Comment by Twoody** [comment]
Any update from anyone on what happened? Does this warrant hosting some form of `https://js.stripe.com/v3` that is considered stable, and a chore then exists to update that stable link over time?

**Comment by softbeehive** [comment]
@Twoody 
Though, Stripe has been far from perfect in communication, I don't recommend hosting stripe.js script yourself. As far as I know, it _breaks PCI compliance_.

I contacted Stripe regarding financial support of the open source ecosystem around their services. In that email I mentioned this issue and asked for clarification.

They responded that my request is not clear. Long story short: not very cool chatting with AI bot disguised as developer support. After I elaborated, they told me to contact the media department 😄 

I had an impression, walls Stripe put around its engineers are too high. Hard to talk to someone who gets to the core of the problem. Anyway, the beauty of open source - we are free to talk about topics others prefer to keep their mouth shut.
